src/main/java/org/humancellatlas/ingest/state/StateMachineConfiguration.java

```diff
@@ -185,6 +188,15 @@ public void configure(StateMachineTransitionConfigurer<SubmissionState, Submissi
                 .event(ALL_TASKS_COMPLETE);
     }
 
+    private Action<SubmissionState, SubmissionEvent> resetTracker(String tracker) {
+        return context -> {
+            DocumentTracker documentTracker = (DocumentTracker) context.getExtendedState().getVariables().get(tracker);
+            if (documentTracker != null) {
```

Comment:
This defensiveness is redundant.
You could do just:

```java
context.getExtendedState().getVariables().remove(tracker)
```
See the doc for [Map.remove(key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#remove-java.lang.Object-)

Reply:
done, thanks! good catch!